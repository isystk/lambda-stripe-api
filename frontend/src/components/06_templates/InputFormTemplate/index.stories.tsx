import { Meta, Story } from '@storybook/react'
import React from 'react'
import InputFormTemplate from './index'
import MainService from '@/services/main'

export default {
  title: '06_templates/InputFormTemplate',
  component: InputFormTemplate,
} as Meta

const Template: Story = () => {
  const main = new MainService(() => ({}))
  const props: InputFormTemplate = { main }
  return <InputFormTemplate {...props} />
}

export const Primary = Template.bind({})
Primary.storyName = 'プライマリ'
Primary.args = {}
