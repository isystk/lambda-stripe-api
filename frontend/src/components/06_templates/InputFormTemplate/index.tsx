import Circles from '@/components/02_interactions/Circles'
import React, { useEffect, FC, useState } from 'react'
import { ContainerProps, WithChildren } from 'types'
import { connect } from '@/components/hoc'
import HtmlSkeleton, { Title } from '@/components/05_layouts/HtmlSkeleton'
import Header from '@/components/04_organisms/Header'
import SideMenu from '@/components/04_organisms/SideMenu'
import MainService from '@/services/main'

/** InputFormTemplateProps Props */
export type InputFormTemplateProps = WithChildren & {
  main: MainService
  title: string
}
/** Presenter Props */
export type PresenterProps = InputFormTemplateProps

/** Presenter Component */
const InputFormTemplatePresenter: FC<PresenterProps> = ({
  children,
  main,
  title,
  isMenuOpen,
  setMenuOpen,
  ...props
}) => (
  <HtmlSkeleton>
    <Title>{title}</Title>
    <Header isMenuOpen={isMenuOpen} setMenuOpen={setMenuOpen} />
    <div className="flex items-center justify-center">
      <Circles>
        <div className="px-1 md:px-32 w-full">
          <div className="py-8">{children}</div>
        </div>
      </Circles>
    </div>
    <div className="md:hidden">
      <SideMenu isMenuOpen={isMenuOpen} setMenuOpen={setMenuOpen} />
    </div>
  </HtmlSkeleton>
)

/** Container Component */
const InputFormTemplateContainer: React.FC<
  ContainerProps<InputFormTemplateProps, PresenterProps>
> = ({ presenter, children, ...props }) => {
  const [isMenuOpen, setMenuOpen] = useState(false)
  return presenter({
    children,
    isMenuOpen,
    setMenuOpen,
    ...props,
  })
}

export default connect<InputFormTemplateProps, PresenterProps>(
  'InputFormTemplate',
  InputFormTemplatePresenter,
  InputFormTemplateContainer
)
