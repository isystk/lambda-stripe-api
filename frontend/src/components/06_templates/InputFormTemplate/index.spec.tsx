import React from 'react'
import renderer from 'react-test-renderer'
import InputFormTemplate, { InputFormTemplateProps } from './index'

describe('InputFormTemplate', () => {
  it('Match Snapshot', () => {
    const props: InputFormTemplateProps = { title: 'サンプル' }
    const component = renderer.create(<InputFormTemplate {...props} />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
