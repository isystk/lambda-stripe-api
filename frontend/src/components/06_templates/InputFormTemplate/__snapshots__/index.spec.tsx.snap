// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputFormTemplate Match Snapshot 1`] = `
[
  <header
    className="css-0 flex justify-between items-center h-16 px-4 py-4 sm:px-8 bg-main w-full"
  >
    <a
      className="css-0"
      href="http://localhost:9000/"
    >
      <div
        className="w-20 md:w-40"
      >
        <img
          className="css-s08p0c "
          src="/images/logo.png"
        />
      </div>
    </a>
    <div
      className="flex pr-3 z-50 md:hidden "
    >
      <button
        className="space-y-2 css-jgxxba"
        onClick={[Function]}
      >
        <div
          className="w-6 h-0.5 bg-black"
        />
        <div
          className="w-6 h-0.5 bg-black"
        />
        <div
          className="w-6 h-0.5 bg-black"
        />
      </button>
    </div>
    <nav
      className="hidden sm:block"
    >
      <ul
        className="flex gap-6"
      >
        <li>
          <a
            href="http://localhost:9000/"
            rel=""
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="http://localhost:9000/product/prod_NpvV9ohJIlgElI/payment"
            rel=""
          >
            Rates
          </a>
        </li>
        <li>
          <a
            href="https://blog.isystk.com/contact/"
            rel="noreferrer"
            target="_blank"
          >
            Contact Us
          </a>
        </li>
      </ul>
    </nav>
  </header>,
  <div
    className="flex items-center justify-center"
  >
    <div
      className="css-1wwfi4"
    >
      <ul
        className="css-10ffb9g"
      >
        <li
          className="css-i6ammj"
        />
        <li
          className="css-l07tar"
        />
        <li
          className="css-1hre2z7"
        />
        <li
          className="css-p2xyi8"
        />
        <li
          className="css-u8edpl"
        />
        <li
          className="css-ik3prs"
        />
        <li
          className="css-1wh69e5"
        />
        <li
          className="css-1cg3at6"
        />
        <li
          className="css-1iaru34"
        />
        <li
          className="css-1io9drp"
        />
      </ul>
      <div
        className="relative"
      >
        <div
          className="px-1 md:px-32 w-full"
        >
          <div
            className="py-8"
          />
        </div>
      </div>
    </div>
  </div>,
  <div
    className="md:hidden"
  >
    <div
      className="css-0 fixed top-0 right-0 z-30 bg-base h-screen w-64 css-hc9ocb"
    >
      <div
        className="mt-24"
      />
      <ul
        className="list-none"
      >
        <li
          className="py-4"
        >
          <a
            className="block text-gray-700 font-bold whitespace-nowrap px-4"
            href="http://localhost:9000/"
            rel=""
          >
            Home
          </a>
        </li>
        <li
          className="py-4"
        >
          <a
            className="block text-gray-700 font-bold whitespace-nowrap px-4"
            href="http://localhost:9000/product/prod_NpvV9ohJIlgElI/payment"
            rel=""
          >
            Rates
          </a>
        </li>
        <li
          className="py-4"
        >
          <a
            className="block text-gray-700 font-bold whitespace-nowrap px-4"
            href="https://blog.isystk.com/contact/"
            rel="noreferrer"
            target="_blank"
          >
            Contact Us
          </a>
        </li>
      </ul>
    </div>
    <div
      className="hidden z-20 bg-black opacity-50 fixed top-0 w-full h-screen"
      onClick={[Function]}
    />
  </div>,
]
`;
